### ISSUE TITLE
Accidental strict dep on LatencyUtils

### ISSUE THREAD

**Issue by codefromthecrypt** [micrometer-metrics/micrometer#1599]
It seems there are some guards to try to not use LatencyUtils unless pause detection is in use. However, there's still a compile dep.

```
[INFO] Caused by: java.lang.ClassNotFoundException: org.LatencyUtils.IntervalEstimator
[INFO] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:471) ~[?:?]
[INFO] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:588) ~[?:?]
[INFO] 	at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:93) ~[zipkin-server-2.16.3-SNAPSHOT-slim.jar:?]
[INFO] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
[INFO] 	at io.micrometer.prometheus.PrometheusMeterRegistry.newTimer(PrometheusMeterRegistry.java:176) ~[micrometer-registry-prometheus-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.MeterRegistry.lambda$timer$2(MeterRegistry.java:271) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.MeterRegistry.getOrCreateMeter(MeterRegistry.java:576) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.MeterRegistry.registerMeterIfNecessary(MeterRegistry.java:529) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.MeterRegistry.timer(MeterRegistry.java:269) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.Timer$Builder.register(Timer.java:477) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.MeterRegistry.timer(MeterRegistry.java:396) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.internal.TimedExecutorService.<init>(TimedExecutorService.java:40) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.binder.jvm.ExecutorServiceMetrics.monitor(ExecutorServiceMetrics.java:94) ~[micrometer-core-1.2.1.jar!/:1.2.1]
[INFO] 	at io.micrometer.core.instrument.binder.jvm.ExecutorServiceMetrics.monitor(ExecutorServiceMetrics.java:107) ~[micrometer-core-1.2.1.jar!/:1.2.1]
```

**Comment by adriancole** [comment]
here's an example of another project working around a similar accidental strict dep https://github.com/line/armeria/pull/2107

**Comment by adriancole** [comment]
It looks like HdrHistogram is also effectively mandatory, It would be nice for these to be optional.

```
java.lang.NoClassDefFoundError: org/HdrHistogram/DoubleRecorder
	at io.micrometer.core.instrument.distribution.TimeWindowPercentileHistogram.<init>(TimeWindowPercentileHistogram.java:37) ~[micrometer-core-1.2.1.jar!/:1.2.1]
	at io.micrometer.core.instrument.AbstractTimer.<init>(AbstractTimer.java:85) ~[micrometer-core-1.2.1.jar!/:1.2.1]
	at io.micrometer.prometheus.PrometheusTimer.<init>(PrometheusTimer.java:40) ~[micrometer-registry-prometheus-1.2.1.jar!/:1.2.1]
	at io.micrometer.prometheus.PrometheusMeterRegistry.newTimer(PrometheusMeterRegistry.java:176) ~[micrometer-registry-prometheus-1.2.1.jar!/:1.2.1]
```

**Comment by adriangonz** [comment]
Is this task currently prioritized for the next release? Is there any time estimate of when it would be ready?

If I can get a "deep dive tour" around the relevant areas of the codebase I would be happy to jump on this one!

**Comment by sheepdreamofandroids** [comment]
This problem still exists. At least declare the dependencies until you find a good way to make them optional.

**Comment by shakuzen** [comment]
LatencyUtils is currently a declared runtime dependency in micrometer-core. If you are using the default configuration (NoPauseDetector) then you should be able to exclude the LatencyUtils dependency in your application. In other words, it should be optional although it is declared. I tried excluding it and ran into an issue at runtime. I think we can easily improve that, at least, without affecting backwards compatibility.

**Comment by shakuzen** [comment]
Resolved by #3262

**Comment by shakuzen** [comment]
If you are not configuring a pause detector, you can exclude the LatencyUtils dependency.
If you are not configuring client-side percentiles, you can exclude the HdrHistogram dependency.
